colearn_examples/grpc/mnist_grpc.py

```diff
@@ -112,19 +119,19 @@ def prepare_learner(data_loaders: Tuple[PrefetchDataset, PrefetchDataset, Prefet
     x = tf.keras.layers.MaxPooling2D((2, 2), name="pool3")(x)
     x = tf.keras.layers.Flatten(name="flatten")(x)
     x = tf.keras.layers.Dense(64, activation="relu", name="fc1")(x)
-    x = tf.keras.layers.Dense(10, activation="softmax", name="fc2")(x)
+    x = tf.keras.layers.Dense(n_classes, activation="softmax", name="fc2")(x)
     model = tf.keras.Model(inputs=input_img, outputs=x)
 
     opt = optimizer(lr=learning_rate)
-    model.compile(loss=loss, metrics=[tf.keras.metrics.SparseCategoricalAccuracy()], optimizer=opt)
+    model.compile(loss=loss, metrics=metric_list, optimizer=opt)
 
     learner = KerasLearner(
         model=model,
         train_loader=data_loaders[0],
         vote_loader=data_loaders[1],
         test_loader=data_loaders[2],
-        criterion="sparse_categorical_accuracy",
-        minimise_criterion=False,
+        criterion="loss",
```

Comment:
to avoid having to set both `metric_list` and `criterion` separately would it be reasonable to :
- always expect the vote criterion to be part of the metric_list?
- simply passing `metric_list[0]` as the `criterion`?

Even more, how about passing to whole `metric_list` to the `KerasLearner` as a substitute for `criterion`?

Reply:
Metrics and loss are two different things. The model should try to minimize or maximize the loss function. The metrics are then used for checking how well the model did on these metrics. There are cases when the loss function is a metric that's true. But I would really keep it separate to avoid confusion / mixture of those two aspects.